### PR TITLE
AF-1962: Fixing Docks for Form Editor

### PR DESCRIPTION
**Pull request by jesuino**
The issue from AF-1962 is related to how it was managing docks before. Instead creating a dock and managing it manually, this PR changes it to use a FormEditorDocksHandler which hides and shows docks if the layout editor is focused or not.

### CODE REVIEW

**Comment by jesuino** [comment]
Hello @paulovmr  and @barboras7 here's the PR for AF-1962, would you please review?

**Comment by jesuino** [comment]
Jenkins execute full downstream build

**Comment by jesuino** [comment]
Jenkins execute full downstream build

**Comment by barboras7** [comment]
Jenkins execute full downstream build

File: kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/events/LayoutEditorLostFocusEvent.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
```
Comment (barboras7) [inline kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/events/LayoutEditorLostFocusEvent.java]: 2019 :)

**Pull request merged**